flag/flags.go

```diff
@@ -0,0 +1,12 @@
+package flag
+
+// Flags provides the ability to create and communicate feature flags.
+type Flags uint64
+
+// Flags is a bitset
+const (
+	Debug Flags = 1 << iota
+	SamplingSet
```

Comment:
thanks!  I've not used proto3 and now learned something. maybe add this to
the comment? (that the isRoot flag helps disambiguate parent id in encoding
formats that cannot represent optional scalars)